backend/estimate-service/src/main/java/org/egov/service/NotificationService.java

```diff
@@ -0,0 +1,262 @@
+
+        return smsDetails;
+    }
+
+    private Map<String, String> getProjectDetails(EstimateRequest request) {
+        Object projectRes = projectServiceUtil.getProjectDetails(request);
+
+        Map<String, String> projectDetails = new HashMap<>();
+        List<String> projectNames = new ArrayList<>();
+        List<String> boundaries = new ArrayList<>();
+        List<String> boundaryTypes = new ArrayList<>();
+        try {
+            projectNames = JsonPath.read(projectRes, PROJECT_NAME_CODE);
+            boundaries = JsonPath.read(projectRes, PROJECT_BOUNDARY_CODE);
+            boundaryTypes = JsonPath.read(projectRes, PROJECT_BOUNDARY_TYPE_CODE);
+
+        } catch (Exception e) {
+            throw new CustomException("PARSING_ERROR", "Failed to parse HRMS response");
+        }
+
+        projectDetails.put("projectName", projectNames.get(0));
+        projectDetails.put("boundary", boundaries.get(0));
+        projectDetails.put("boundaryType", boundaryTypes.get(0));
+
+        return projectDetails;
+    }
+
+    private String getMessage(EstimateRequest request) {
+        Workflow workflow = request.getWorkflow();
+        String message = null;
+
+        if ("REJECT".equalsIgnoreCase(workflow.getAction())) {
+            message = getMessage(request, ESTIMATE_REJECT_LOCALIZATION_CODE);
+        } else if ("APPROVE".equalsIgnoreCase(workflow.getAction())) {
+            message = getMessage(request, ESTIMATE_APPROVE_LOCALIZATION_CODE);
+        }
+
+        return message;
+    }
+
+    /**
+     * Gets the message from localization
+     *
+     * @param request
+     * @param msgCode
+     * @return
+     */
+    public String getMessage(EstimateRequest request, String msgCode) {
+        String rootTenantId = request.getEstimate().getTenantId().split("\\.")[0];
+        RequestInfo requestInfo = request.getRequestInfo();
+        Map<String, Map<String, String>> localizedMessageMap = getLocalisedMessages(requestInfo, rootTenantId,
+                EstimateServiceConstant.ESTIMATE_NOTIFICATION_ENG_LOCALE_CODE, EstimateServiceConstant.ESTIMATE_MODULE_CODE);
+        return localizedMessageMap.get(EstimateServiceConstant.ESTIMATE_NOTIFICATION_ENG_LOCALE_CODE + "|" + rootTenantId).get(msgCode);
+    }
+
+    /**
+     * Builds msg based on the format
+     *
+     * @param estimate
+     * @param message
+     * @param userDetailsForSMS
+     * @return
+     */
+    public String buildMessageForRejectAction(Estimate estimate, Map<String, String> userDetailsForSMS, String message) {
+        message = message.replace("{estimate_no}", estimate.getEstimateNumber())
+                .replace("{project_name}", userDetailsForSMS.get("projectName"))
+                .replace("{location}", userDetailsForSMS.get("locationName"))
+                .replace("{username}", userDetailsForSMS.get("userName"))
+                .replace("{designation}", userDetailsForSMS.get("designation"));
+        return message;
+    }
+
```

Comment:
"{estimate_no}" , "{project_name}" etc can be declared as constant. Because they are used multiple places

Reply:
Keeping it generic for now to accomodate for message template changes later as suggested by CK